lib/stm32/f4/rcc.c

```diff
@@ -743,6 +727,8 @@ void rcc_clock_setup_hse_3v3(const struct rcc_clock_scale *clock)
 	rcc_wait_for_osc_ready(RCC_PLL);
 
 	/* Configure flash settings. */
+	flash_dcache_enable();
+	flash_icache_enable();
```

Comment:
Good point. Would it be acceptable to have a check for those bits there and then call the associated flash_Xcache_enable() function if needed? That way the config options could still be passed in the struct but would not rely on flash_set_ws().